### PR TITLE
MXRT600: Add support for SPI

### PR DESCRIPTION
**Pull request by mmahadevan108**
Enable access to the arduino pins on the mxrt600 evk board. Passed `tests/drivers/spi/spi_loopback` test, below are the test results:

```
*** Booting Zephyr OS build zephyr-v2.3.0-1884-g6e443da2ad25  ***
Running test suite test_spi
===================================================================
START - test_spi_loopback
I: SPI test on buffers TX/RX 0x30000131/0x300007a6
I: SPI test slow config
I: Start complete multiple
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start async call
I: Passed
I: SPI test fast config
I: Start complete multiple
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Start every 4
I: Passed
I: Start async call
I: Passed
I: Start complete loop
I: CS control inhibited (no GPIO device)
I: Passed
I: Start complete loop
I: CS control inhibited (no GPIO device)
I: Passed
I: All tx/rx passed
 PASS - test_spi_loopback
===================================================================
Test suite test_spi succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
```

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @MaureenHelm @agansari @dleach02

**Review by MaureenHelm — APPROVED** [review]

File: boards/arm/mimxrt685_evk/pinmux.c [hunk 1]
```diff
@@ -192,6 +192,92 @@ static int mimxrt685_evk_pinmux_init(struct device *dev)
...
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+	/* PORT1 PIN5 (coords: J16) is configured as FC5_RXD_SDA_MOSI_DATA */
+	IOPCTL_PinMuxSet(IOPCTL, 1U, 5U, port1_pin5_config);
+
+	const uint32_t port1_pin6_config = (/* Pin is configured as FC5_SSEL0 */
```
Comment (agansari) [inline boards/arm/mimxrt685_evk/pinmux.c:259]: If you want the ability to have either Flexcomm's SS or GPIO CS as mentioned in dts file make a change similar to the one below:

https://github.com/zephyrproject-rtos/zephyr/blob/c5d85e175fae85dce4236d20a92717e084b57709/boards/arm/lpcxpresso55s69/pinmux.c#L128-L138

File: dts/arm/nxp/nxp_rt6xx_common.dtsi [hunk 2]
```diff
@@ -138,6 +138,22 @@
 		label = "FLEXCOMM_7";
 		status = "disabled";
 	};
+
+	hs_lspi: spi@126000 {
+		compatible = "nxp,lpc-spi";
+		/* Enabling cs-gpios below will allow using GPIO CS,
+		 rather than Flexcomm SS */
+		/* cs-gpios = <&gpio1 14 GPIO_ACTIVE_LOW>,
```
Comment (agansari) [inline dts/arm/nxp/nxp_rt6xx_common.dtsi:146]: see comment in pinmux.c

**Review by agansari — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt685_evk/pinmux.c (hunk 1 above)
Comment (mmahadevan108) [reply boards/arm/mimxrt685_evk/pinmux.c:259]: Thank you, this is good to know. In this case, the configuration is for FlexComm5 and not HS_LSPI, hence we need not apply this.

File: dts/arm/nxp/nxp_rt6xx_common.dtsi (hunk 2 above)
Comment (mmahadevan108) [reply dts/arm/nxp/nxp_rt6xx_common.dtsi:146]: Thank you, this is good to know. In this case, the configuration is for FlexComm5 and not HS_LSPI, hence we need not apply this.

**Comment by mmahadevan108** [comment]
Anything else needed for this PR?

**Comment by agansari** [comment]
> Anything else needed for this PR?

mmahadevan108 I don't see the issue I raised addressed.

**Comment by mmahadevan108** [comment]
> > Anything else needed for this PR?
> 
> mmahadevan108 I don't see the issue I raised addressed.

I replied earlier that this cannot be applied here as the #define is for a FLEXCOMM SPI and not HS_SPI

File: dts/arm/nxp/nxp_rt6xx_common.dtsi (hunk 2 above)
Comment (agansari) [reply dts/arm/nxp/nxp_rt6xx_common.dtsi:146]: @mmahadevan108 now i see there are different spi ports.

**Review by agansari — APPROVED** [review]

**Pull request merged**